test/src/unit/modules/ProjectList.test.js

```diff
@@ -0,0 +1,297 @@
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            
+            const updatedProject = { ...project };
+            updatedProject.name = 'NEW';
+
+            project.should.haveOwnProperty('name').and.equal('newdummyproject');
+            updatedProject.should.haveOwnProperty('name').and.equal('NEW');
+
+            const returnedProject = await projectList.updateProject(updatedProject);
+            returnedProject.should.haveOwnProperty('name').and.equal('NEW');
+            const returnedProjectInfPath = path.join(global.codewind.CODEWIND_WORKSPACE, '.projects', `${returnedProject.projectID}.inf`);            
+            fs.existsSync(returnedProjectInfPath).should.be.true;
+        });
+        it('Updates a project\'s host to "ingress_host" and appStatus to "started"', async() => {
+            const projectList = new ProjectList();
+            const project = new Project({ name: 'newdummyproject' }, global.codewind.CODEWIND_WORKSPACE);
+            projectList.addProject(project);
+            // CHE_INGRESS_HOST is set when appStatus is set to started
+            process.env.CHE_INGRESS_HOST = 'ingress_host';
+            
+            const updatedProject = { ...project };
+            updatedProject.appStatus = 'started';
+
+            project.should.not.haveOwnProperty('appStatus');
+            project.should.haveOwnProperty('host').and.equal('');
+            updatedProject.should.haveOwnProperty('appStatus').and.equal('started');
+            updatedProject.should.haveOwnProperty('host').and.equal('');
+
+            const returnedProject = await projectList.updateProject(updatedProject);
+            returnedProject.should.haveOwnProperty('appStatus').and.equal('started');
+            returnedProject.should.haveOwnProperty('host').and.equal('ingress_host');
+            const returnedProjectInfPath = path.join(global.codewind.CODEWIND_WORKSPACE, '.projects', `${returnedProject.projectID}.inf`);            
+            fs.existsSync(returnedProjectInfPath).should.be.true;
+        });
+    });
+    describe('reloadSettingsFile(reloadProject)', () => {
+        it('Errors as project does not have a projectID', () => {
+            const projectList = new ProjectList();
+            const func = () => projectList.reloadSettingsFile({});
+            return func().should.be.rejectedWith('MALFORMED', null, 'Project does not have a project ID');
+        });
+        it('Errors as project is not a member of the ProjectList', () => {
+            const projectList = new ProjectList();
+            const func = () => projectList.reloadSettingsFile({ projectID: '123' });
+            return func().should.be.rejectedWith('NOT_FOUND', '123');
+        });
+        it('Reloads a project by reading from its .cw-settings file and picks up a new field', async() => {
```

Comment:
Can this message refer to the new setting coming from .cw-settings? If I've understood it correctly